sycl/source/handler.cpp

```diff
@@ -543,8 +543,6 @@ event handler::finalize() {
         NewEvent->setSubmissionTime();
 
         EnqueueKernel();
-        if (NewEvent->isHost() || NewEvent->getHandle() == nullptr)
```

Comment:
Does it make sense to add assert for both conditions or at least `Handle != nullptr`?